build.sh

```diff
@@ -1,23 +1,71 @@
 #!/bin/bash
 # the path to the ComputeCPP package root directory, e.g. /home/user/ComputeCpp-CE-0.1-Linux/
-PACKAGE_ROOT=$1
 
-function install_gmock  {
-  mkdir -p external && pushd external
-  git clone git@github.com:google/googletest.git
-  pushd googletest/googlemock/make && make
-  popd 
-  popd
-}
+# Useless to go on when an error occurs
+set -o errexit
+
+# How to use build.sh to compile SyclParallelSTL with ComputeCpp ?
+# ./build.sh "path/to/ComputeCpp"
+#
+# How to use build.sh to compile SyclParallelSTL with triSYCL ?
+# ./build.sh --trisycl "path/to/triSYCL" "path/to/compute"
+#
+# paths can be relative ones, this script will make them absoulte before to send them to CMake
+#
+# WARNING/TODO: setting the "path/to/compute" in triSYCL mode is not an option yet
+#
+
+if [ $1 == "--trisycl" ]
+then
+	shift
+	echo "build.sh enter mode: triSYCL"
+	CMAKE_ARGS="$CMAKE_ARGS -DUSE_COMPUTECPP=OFF"
+	if [ $1 ] # $1 is actually $2
+	then
+		CMAKE_ARGS="$CMAKE_ARGS -DTRISYCL_INCLUDE_DIR=$(readlink -f $1)/include"
+		shift
+	fi
+	if [ $1 ] # $1 is actually $3
+	then
+		CMAKE_ARGS="$CMAKE_ARGS -DBOOST_COMPUTE_INCLUDE_DIR=$(readlink -f $1)/include"
+		shift
+	fi
+else
+	echo "build.sh enter mode: ComputeCpp"
+	CMAKE_ARGS="$CMAKE_ARGS -DCOMPUTECPP_PACKAGE_ROOT_DIR=$(readlink -f $1)"
+	shift
+fi
+NPROC=$(nproc)
+
+function install_gmock  {(
+  #REPO="git@github.com:google/googletest.git"
+  REPO="https://github.com/google/googletest.git"
+  mkdir -p external
+  cd external
+  if [ -d googletest ]
+  then
+	  cd googletest
+	  git pull
+  else
+	  git clone $REPO
+	  cd googletest
+  fi
+  cd googlemock/make
+  make -j$NPROC
+)}
 
 function configure  {
-  mkdir -p build && pushd build 
-  cmake .. -DCOMPUTECPP_PACKAGE_ROOT_DIR=$PACKAGE_ROOT  -DPARALLEL_STL_BENCHMARKS=ON
+  mkdir -p build && pushd build
+  if [ $@ ]
+  then
+	  echo "cmake additionnal args: $@"
+  fi
+  cmake .. $CMAKE_ARGS -DPARALLEL_STL_BENCHMARKS=ON $@
```

Comment:
How is it possible to add more arguments to cmake without setting the triSYCL and Boost.Compute path?
The more I think to it and the less I am kind of adding all that stuff to specify where triSYCL and Boost.Compute are...